rust/cubesql/cubesql/src/compile/engine/udf/common.rs

```diff
@@ -3148,6 +3155,155 @@ pub fn create_pg_get_serial_sequence_udf() -> ScalarUDF {
+            ));
+        }
+
+        let format_strings = downcast_string_arg!(&args[0], "format_str", i32);
+        let mut builder = StringBuilder::new(format_strings.len());
+
+        for i in 0..format_strings.len() {
+            if format_strings.is_null(i) {
+                builder.append_null()?;
+                continue;
+            }
+
+            let format_str = format_strings.value(i);
+            let mut result = String::new();
+            let mut format_chars = format_str.chars().peekable();
+            let mut arg_index = 1; // Start from first argument after format string
+
+            while let Some(c) = format_chars.next() {
+                if c != '%' {
+                    result.push(c);
+                    continue;
+                }
+
+                match format_chars.next() {
+                    Some('I') => {
+                        // Handle %I - SQL identifier
+                        if arg_index >= args.len() {
+                            return Err(DataFusionError::Execution(
+                                "Not enough arguments for format string".to_string(),
+                            ));
+                        }
+
+                        let arg = &args[arg_index];
+                        let value = match arg.data_type() {
+                            DataType::Utf8 => {
+                                let str_arr = downcast_string_arg!(arg, "arg", i32);
+                                if str_arr.is_null(i) {
+                                    return Err(DataFusionError::Execution(
+                                        "NULL values cannot be formatted as identifiers"
+                                            .to_string(),
+                                    ));
+                                }
+                                str_arr.value(i).to_string()
+                            }
+                            _ => {
+                                // For other types, try to convert to string
+                                let str_arr = cast(&arg, &DataType::Utf8)?;
+                                let str_arr =
+                                    str_arr.as_any().downcast_ref::<StringArray>().unwrap();
+                                if str_arr.is_null(i) {
+                                    return Err(DataFusionError::Execution(
+                                        "NULL values cannot be formatted as identifiers"
+                                            .to_string(),
+                                    ));
+                                }
+                                str_arr.value(i).to_string()
+                            }
+                        };
+
+                        // Quote identifier if necessary
+                        let needs_quoting = !value
+                            .chars()
+                            .all(|c| c.is_ascii_lowercase() || c.is_ascii_digit() || c == '_')
```

Comment:
This is a bit wrong, because identifiers `1`  and `1a` need quoting, but `a1` - does not.